app/src/main/java/com/example/splendormobilegame/activities/GameActivity/TurnController.java

```diff
@@ -25,8 +25,8 @@ protected TurnController(T activity) {
     }
 
     public void endTurn() {
-        // Maybe you want to check some things before sending request
-        this.sendRequestToEndTurn();
+        if(Model.getRoom().getGame().getWhosTurn()==Model.getRoom().getUserByUuid(Model.getUserUuid())){ this.sendRequestToEndTurn(); }
+        else{ Log.i("UserReaction", "Player verification step failed"); }
```

Comment:
Please improve this log message to include the following information:
- Mention that the event occurred in the TurnController.
- Specify that it happened while the user attempted to end the turn.
- Clarify that the request was not sent as a result of this event.